core/foundation/test/TypeTraits.cxx

```diff
@@ -0,0 +1,143 @@
+#include "ROOT/TypeTraits.hxx"
```

Comment:
Can we please rename the test file either TypeTraits.test.cxx (or testTypeTrait.cxx) to distinguish it (beyhond just the directory) from an implementation file?  Thanks.